### ISSUE TITLE
ChatToolChoiceType enum is missing REQUIRED value

### ISSUE THREAD

**Issue by the-gigi** [sashirestela/simple-openai#116]
The OpenAI API allows to specify `tool_choice: "required"`
See https://platform.openai.com/docs/guides/function-calling/function-calling-behavior

The simple-openai enum offers only "none" and "auto"

```
public enum ChatToolChoiceType {

    @JsonProperty("none")
    NONE,

    @JsonProperty("auto")
    AUTO;

}
```

**Comment by sashirestela** [comment]
That feature was recently added and it will be part of the release 3.0.0

See: https://platform.openai.com/docs/changelog/apr-29th-2024

**Comment by the-gigi** [comment]
This issue will be resolved in 3.0.0.
